src/mango_native_proc.erl

```diff
@@ -283,11 +283,20 @@ get_text_field_type(_) ->
 
 make_text_field(TAcc, Type, Value) ->
     FieldName = make_text_field_name(TAcc#tacc.path, Type),
-    Fields = TAcc#tacc.fields,
-    case Fields == all_fields orelse lists:member(FieldName, Fields) of
+    Fields = case TAcc#tacc.fields of
+        Fields0 when is_list(Fields0) ->
+            % for field names such as "name\\.first"
+            PFields = [mango_doc:parse_field(F) || F <- Fields0],
+            [iolist_to_binary(mango_util:join(".", P)) || {ok, P} <- PFields];
```

Comment:
I think you're missing a level of unescaping here. The example you gave is kind of odd as well as it seems to be referencing user escapes but from database data (which isn't impossible, its just complicating things a bit).

For, instance, consider these three documents:

    {"_id":"a", "name":{"first":"Tony"}}
    {"_id": "b", "name.first":"Tony"}
    {"_id":"c", "name\.first":"Tony"}

Note for that last example I'm ignoring any sort of escaping that might be required based on where the string is defined. For instance, in Erlang binaries it would be written <<"name\\.first">> as '\' is an escape character in those other contexts.

The value "Tony" would then be in the database with three different fields:

    name.first_3astring
    name_2efirst_3astring
    name_5c_2efirst_3astring

The contents of a database are *not* passed through the user escaping functions. Its important to note that we don't care that the pattern '\.' happens in the document key. Its escaped as two characters just like anything else.

Now on the user input side of things when we want to address a field that contained a '.' character we have to ensure that its escaped correctly based on what the client gave us. For instance, queries corresponding the above would look like such:

    name.first
    name\.first
    name\\.first

Again, my comment about escaping is that these examples do **not** account for how this might be written in other contexts. For instance, if those were written in JSON they'd double up the '\' characters because JSON uses the `\' as its own escape.

The only thing the client should have to care about is if they have a period in a doc key, they have to escape it using the \ escape and they may have to escape that \ in their source code due to escaping of the escape.